### DISCUSSION TITLE
What exact command creates the spv files in vsgExamples?

### DISCUSSION THREAD

**Discussion by dov** [vsg-dev/VulkanSceneGraph#1237] [General]
I wonder what exactly are in the spv files in vsgExamples. The relation between the shader sources and the spv files isn't always obvious to me.

I assume that the compilation command is simply:

```
glslang -o foo_frag.spv -V foo.frag 
```

But perhaps there flags for some of the files. Why at all are the spv files in git instead of being built at compilation time? 

An example of a file whose source code isn't obvious is `comp.spv` which is loaded in `vsgcompute.cpp`.

**Comment by robertosfield** [comment]
You can use vsgconv from vsgXchange if you have built the VSG with glslang support.

The vsgshaderset example in vsgExamples also does shader compilation but in the context of VSG native datastructures and file formats.  Again dependent on VSG glslang support.

But you can also just have the VSG compile the GLSL into SPV at runtime so no acute need to generated SPV files.
